workflow/controller/operator.go

```diff
@@ -1210,7 +1210,7 @@ func getExitCode(pod *apiv1.Pod) *int32 {
 
 func (woc *wfOperationCtx) cleanUpPod(pod *apiv1.Pod) {
 	for _, c := range pod.Status.ContainerStatuses {
-		if c.Name == common.WaitContainerName && c.State.Terminated == nil {
+		if (c.Name == common.WaitContainerName || c.Name == common.MainContainerName) && c.State.Terminated == nil {
```

Comment:
the goal of this code is to kill sidecar container, but I think checking just the name is wrong, I think you need to use `IsMainContainerName`

Reply:
Updated.